libraries/esp8266/examples/NTP-TZ-DST/NTP-TZ-DST.ino

```diff
@@ -1,80 +1,148 @@
 /*
-  NTP-TZ-DST
+  NTP-TZ-DST (v2)
   NetWork Time Protocol - Time Zone - Daylight Saving Time
 
-  This example shows how to read and set time,
-  and how to use NTP (set NTP0_OR_LOCAL1 to 0 below)
-  or an external RTC (set NTP0_OR_LOCAL1 to 1 below)
-
-  TZ and DST below have to be manually set
-  according to your local settings.
+  This example shows:
+  - how to read and set time
+  - how to set timezone per country/city
+  - how is local time automatically handled per official timezone definitions
 
   This example code is in the public domain.
 */
 
-#include <ESP8266WiFi.h>
-#include <time.h>                       // time() ctime()
-#include <sys/time.h>                   // struct timeval
-#include <coredecls.h>                  // settimeofday_cb()
-
-////////////////////////////////////////////////////////
 
 #ifndef STASSID
 #define STASSID "your-ssid"
 #define STAPSK  "your-password"
 #endif
 
-#define SSID            STASSID
-#define SSIDPWD         STAPSK
-#define TZ              1       // (utc+) TZ in hours
-#define DST_MN          60      // use 60mn for summer time in some countries
+// initial time (possibly given by an external RTC)
+#define RTC_UTC_TEST 1510592825 // 1510592825 = Monday 13 November 2017 17:07:05 UTC
 
-#define NTP0_OR_LOCAL1  1       // 0:use NTP  1:fake external RTC
-#define RTC_TEST     1510592825 // 1510592825 = Monday 13 November 2017 17:07:05 UTC
+
+// This database is autogenerated from IANA timezone database
+//    https://www.iana.org/time-zones
+// and can be updated on demand in this repository
+#include <TZ.h>
+
+// "TZ_" macros follow DST change across seasons without source code change
+// check for your nearest city in TZ.h
+
+// select espressif headquarter TZ
+#define MYTZ TZ_Asia_Shanghai
+
+// example for "Not Only Whole Hours" timezones:
+// Kolkata/Calcutta is shifted by 30mn
+//#define MYTZ TZ_Asia_Kolkata
+
+// example to watch automatic local time time adjustment on Summer/Winter change
+// (exactly 182 days (~6 month) are regularly added or substracted in the example)
+//#define MYTZ TZ_Europe_Paris
```

Comment:
Make this default.  China doesn't do daylight savings, so there is no difference before/after in the script output.